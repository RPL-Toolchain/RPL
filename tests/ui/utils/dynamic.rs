#[rpl::dynamic(
    primary_message = "{$kind} {$lang} pattern",
    args(kind = "Dynamic", lang = "RPL"),
    help = "You can use `#[rpl::dynamic]` to create a customizable lint.",
    note = "This is a dynamic RPL pattern, which can be customized during runtime."
)]
fn main() {
    //~^ERROR: Dynamic RPL pattern
    //~|HELP: You can use `#[rpl::dynamic]` to create a customizable lint.
    //~|NOTE: This is a dynamic RPL pattern, which can be customized during runtime.
    // This function is used to trigger the dynamic pattern check.
    // It does not need to do anything.
}
