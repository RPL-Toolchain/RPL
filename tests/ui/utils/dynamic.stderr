error: {$kind} {$lang} pattern
  --> tests/ui/utils/dynamic.rs:7:1
   |
LL | fn main() {
   | ^^^^^^^^^
   |
   = help: You can use `#[rpl::dynamic]` to create a customizable lint.
   = note: This is a dynamic RPL pattern, which can be customized during runtime.
   = note: `#[forbid(rpl::dynamic)]` on by default

error: aborting due to 1 previous error

